# HG changeset patch
# User Rik <rik@octave.org>
# Date 1548362166 28800
#      Thu Jan 24 12:36:06 2019 -0800
# Node ID 7994f91cbdb865bd1b6ec4e54ae41a667b2e9459
# Parent  708d2db72e1c27c2206d12bb98848b12e3b23ba2
Provide correct guess at filename when saving classdef file from GUI (bug #55497).

* file-editor-tab.cc (file_editor_tab::get_function_name):
Add regular expression to search for "classdef NAME".

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -3096,28 +3096,31 @@ namespace octave
   }
 
   QString file_editor_tab::get_function_name (void)
   {
     QRegExp rxfun1 ("^[\t ]*function[^=]+=([^\\(]+)\\([^\\)]*\\)[\t ]*$");
     QRegExp rxfun2 ("^[\t ]*function[\t ]+([^\\(]+)\\([^\\)]*\\)[\t ]*$");
     QRegExp rxfun3 ("^[\t ]*function[^=]+=[\t ]*([^\\s]+)[\t ]*$");
     QRegExp rxfun4 ("^[\t ]*function[\t ]+([^\\s]+)[\t ]*$");
+    QRegExp rxfun5 ("^[\t ]*classdef[\t ]+([^\\s]+)[\t ]*$");
 
     QStringList lines = _edit_area->text ().split ("\n");
 
     for (int i = 0; i < lines.count (); i++)
       {
         if (rxfun1.indexIn (lines.at (i)) != -1)
           return rxfun1.cap (1).remove (QRegExp ("[ \t]*"));
         else if (rxfun2.indexIn (lines.at (i)) != -1)
           return rxfun2.cap (1).remove (QRegExp ("[ \t]*"));
         else if (rxfun3.indexIn (lines.at (i)) != -1)
           return rxfun3.cap (1).remove (QRegExp ("[ \t]*"));
         else if (rxfun4.indexIn (lines.at (i)) != -1)
           return rxfun4.cap (1).remove (QRegExp ("[ \t]*"));
+        else if (rxfun5.indexIn (lines.at (i)) != -1)
+          return rxfun5.cap (1).remove (QRegExp ("[ \t]*"));
       }
 
     return QString ();
   }
 }
 
 #endif
